Automattic/wp-calypso
# Remove auth-code-request-store and replace it with local state

This PR removes the Flux store that's used by the `AuthCodeButton` component to store a success/error result of a SMS verification code request. It's used only by the desktop app when the user has SMS 2FA authentication enabled:

<img width="329" alt="screen shot 2018-07-26 at 12 22 58" src="https://user-images.githubusercontent.com/664258/43262992-460af944-90e2-11e8-980a-3bfd024c6efc.png">

`AuthCodeButton` is the one with "Send code via text message" label. After clicking, the label should change to "Code sent":

<img width="330" alt="screen shot 2018-07-26 at 12 23 24" src="https://user-images.githubusercontent.com/664258/43263040-6a4df2e8-90e2-11e8-8c53-6510680c9c68.png">

And then revert back after 30 seconds.

Because the store is not used by any other component and is not likely to ever be, and because it's so simple, I converted it to the component's local state. It still keeps the component at 120 lines of code.

I think it's a nice example of how we sometimes use advanced tools for trivial problems, where they are not a good fit.

**How to test with Desktop App:**
1. Have an user account that has SMS 2FA enabled (set up in `/me/security/two-step`)
2. Try to login as that user in Desktop app
3. Verify that you see the "Send code via text message" button and that it works

**How to test with local Calypso:**
1. User account with SMS 2FA is needed here as well.
2. Enable `oauth` flag in `config/development.json` (needed both on server and client)
3. Create `config/secrets.json` file per [Desktop app instructions](https://github.com/Automattic/wp-desktop/blob/develop/docs/secrets.md).
4. Start local Calypso (`npm run start`)
5. Go to `/oauth-login`
6. You should see the OAuth version of the login screen and can proceed with step 2 of the Desktop instructions above.
7. Note that you won't be able to actually login in local Calypso, but you should be able to get the 2FA code.
